packages/mdc-ripple/foundation.js

```diff
@@ -145,8 +145,9 @@ export default class MDCRippleFoundation extends MDCFoundation {
     }
 
     activationState.isActivated = true;
+    // If e is null, this is a programmatic activation.
     activationState.activationEvent = e;
-    activationState.wasActivatedByPointer = (
+    activationState.wasActivatedByPointer = e === null ? false : (
```

Comment:
What if instead of checking for `e === null` we added a boolean property `isProgrammatic` to the activationState object, and used that instead? e.g.

```js
activationState.isProgrammatic = e === null;
// ...
const isPointerEvent = !this.activationState_.isProgrammatic && /* ... */;
```